### PR TITLE
Make AnnotationGrabber.METHOD_URL a path (NOT a regex) RMB-808

### PR DESCRIPTION
**Pull request by adamdickmeiss**

### CODE REVIEW

File: domain-models-maven-plugin/src/main/java/org/folio/rest/tools/ClientGenerator.java [hunk 1]
```diff
@@ -334,13 +334,13 @@ public void generateMethodMeta(String methodName, JsonObject params, String url,
     /* Handle place holders in the URL
       * replace {varName} with "+varName+" so that it will be replaced
       * in the url at runtime with the correct values */
-    Matcher m = Pattern.compile("\\{.*?\\}").matcher(url);
-    while(m.find()){
+    Matcher m = Pattern.compile("\\{.*?\\}").matcher("^" + url);
```
Comment (julianladisch) [inline domain-models-maven-plugin/src/main/java/org/folio/rest/tools/ClientGenerator.java:337]: "^" is not needed here.

Comment (adamdickmeiss) [reply domain-models-maven-plugin/src/main/java/org/folio/rest/tools/ClientGenerator.java:337]: Good point!

File: domain-models-maven-plugin/src/main/java/org/folio/rest/tools/ClientGenerator.java [hunk 2]
```diff
@@ -334,13 +334,13 @@ public void generateMethodMeta(String methodName, JsonObject params, String url,
...
+    while (m.find()) {
       String varName = m.group().replace("{","").replace("}", "");
       url = url.replace("{"+varName+"}", "\"+"+varName+"+\"");
     }
 
-    url = "\""+url.substring(1)+"\"+queryParams.toString()";
+    url = "\"" + url + "\"+queryParams.toString()";
```
Comment (julianladisch) [inline domain-models-maven-plugin/src/main/java/org/folio/rest/tools/ClientGenerator.java:343]:
```suggestion
    url = "\"" + 
        Pattern.compile("\\{([^{}/]+)\\}").matcher(url).replaceAll("\"+$1+\"") + 
        "\"+queryParams.toString()";
```

Comment (adamdickmeiss) [reply domain-models-maven-plugin/src/main/java/org/folio/rest/tools/ClientGenerator.java:343]: This is supposed to replace the preceding for loop.. That results in failures.. We need a better test for clientGenerator itself --  the TestResource.java does not expose enough features!

Comment (julianladisch) [reply domain-models-maven-plugin/src/main/java/org/folio/rest/tools/ClientGenerator.java:343]: This should replace all the other Matcher and url code above.

**Review by julianladisch — APPROVED** [review]

File: domain-models-maven-plugin/src/main/java/org/folio/rest/tools/ClientGenerator.java (hunk 2 above)
Comment (adamdickmeiss) [reply domain-models-maven-plugin/src/main/java/org/folio/rest/tools/ClientGenerator.java:343]: OK.. I'll try again.. I like the simplication.

Comment (adamdickmeiss) [reply domain-models-maven-plugin/src/main/java/org/folio/rest/tools/ClientGenerator.java:343]: OK.. I'll try again.. I like the simplification.

Comment (adamdickmeiss) [reply domain-models-maven-plugin/src/main/java/org/folio/rest/tools/ClientGenerator.java]: OK. Good now.

**Pull request merged**